### PR TITLE
Updating previous implementation for user

### PR DESCRIPTION
**Pull request by OleksandRomaniuk**

### CODE REVIEW

**Comment by OleksandRomaniuk** [comment]
@ajax-gnatiuk-s
@ajax-shulepa-a

File: src/main/kotlin/com/example/ajaxproject/service/interfaces/UserService.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    fun updateUser(id: Long, userDTO: UserDTO): User
+
+    fun deleteUser(id: Long)
+
+    fun findUserById(id: Long): User?
+
+    fun findAllUsers(): List<User>
+
+}
```
Comment (ajax-gnatiuk-s) [inline src/main/kotlin/com/example/ajaxproject/service/interfaces/UserService.kt:18]: no new line at the end of file
image


Please setup your IDE to add line break at the end of file automatically

File: src/main/kotlin/com/example/ajaxproject/service/UserServiceImpl.kt [hunk 2]
```diff
@@ -4,36 +4,43 @@ import com.example.ajaxproject.dto.UserDTO
...
+        return userRepository.save(user)
     }
 
-    fun deleteUser(id: Long) {
+    override fun deleteUser(id: Long) {
         userRepository.deleteById(id)
     }
 
+    override fun findUserById(id: Long): User? {
+        return userRepository.findById(id).orElse(null)
```
Comment (ajax-gnatiuk-s) [inline src/main/kotlin/com/example/ajaxproject/service/UserServiceImpl.kt:39]: When you find entity by id and it was not found then throw exception like you did in `#updateUser`

File: src/main/kotlin/com/example/ajaxproject/controller/UserController.kt [hunk 3]
```diff
@@ -44,4 +32,15 @@ class UserController @Autowired constructor(
         return ResponseEntity.noContent().build()
     }
 
+    @GetMapping("/{id}")
+    fun findUserById(@PathVariable id: Long): ResponseEntity<User?> {
+        val user = userService.findUserById(id)
+        return ResponseEntity.ok(user)
+    }
```
Comment (ajax-gnatiuk-s) [inline src/main/kotlin/com/example/ajaxproject/controller/UserController.kt:39]: we expect that user by id is exist. So, we can simplify the function:
```kotlin
    @GetMapping("/{id}")
    fun findUserById(@PathVariable id: Long): User = userService.findUserById(id)
``` 
Isn't it beautiful?

Note: you should have `@ExceptionHandler`(or @ControllerAdvice) that converts exception to 404.
More details:
- your service should throw. Let's say, custom UserNotFoundException.
- then an exception handler intercepts the exception and converts to 404 response

**Review by ajax-gnatiuk-s — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/example/ajaxproject/service/UserServiceImpl.kt (hunk 2 above)
Comment (ajax-shulepa-a) [reply src/main/kotlin/com/example/ajaxproject/service/UserServiceImpl.kt:39]: one more thing. if a method throws an exception on not found, it's better to name it like 'get...'

Comment (OleksandRomaniuk) [reply src/main/kotlin/com/example/ajaxproject/service/UserServiceImpl.kt]: Renamed all find methods to get

File: src/main/kotlin/com/example/ajaxproject/dto/UserDTO.kt [hunk 4]
```diff
@@ -4,9 +4,12 @@ import com.example.ajaxproject.model.enums.Role
...
+
     val secondName: String,
+
     val email: String,
+
     val password: String,
-    val role: Role
-)
+
+    val role: Role)
```
Comment (ajax-gnatiuk-s) [inline src/main/kotlin/com/example/ajaxproject/dto/UserDTO.kt:14]: please change the class declaration to:
```kotlin
class UserDTO(
    val firstName: String,
    val secondName: String,
    val email: String,
    val password: String,
    val role: Role
)
```

File: src/main/kotlin/com/example/ajaxproject/model/ChatMessage.kt [hunk 5]
```diff
@@ -18,6 +23,4 @@ data class ChatMessage(
 
     @JsonProperty("recipientId")
     @Column(name = "recipientId", length = 100)
-    val recipientId: String? = null,
-
-)
\ No newline at end of file
+    val recipientId: String? = null,)
```
Comment (ajax-gnatiuk-s) [inline src/main/kotlin/com/example/ajaxproject/model/ChatMessage.kt:26]: `)` to new line. Please, check all your code

File: src/main/kotlin/com/example/ajaxproject/model/MentorToCategories.kt [hunk 6]
```diff
@@ -1,9 +1,14 @@
...
-
-
+import jakarta.persistence.Column
+import jakarta.persistence.Entity
+import jakarta.persistence.GeneratedValue
+import jakarta.persistence.GenerationType
+import jakarta.persistence.Id
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.ManyToOne
+import jakarta.persistence.Table
```
Comment (ajax-gnatiuk-s) [inline src/main/kotlin/com/example/ajaxproject/model/MentorToCategories.kt:11]: let's separate imports section from code section with one empty line:
```kotlin
package your.package...

import com.first
import com.last

class SomeClass {

}

```

**Review by ajax-gnatiuk-s — CHANGES_REQUESTED** [review]

**Pull request merged**